planner/core/logical_plans.go

```diff
@@ -1071,6 +1071,16 @@ type WindowFrame struct {
 	End   *FrameBound
 }
 
+func (wf *WindowFrame) Clone() *WindowFrame {
+	cloned := new(WindowFrame)
+	*cloned = *wf
+
+	cloned.Start = wf.Start.Clone()
+	cloned.End = wf.End.Clone()
+
+	return cloned
+}
```

Comment:
Why we need to clone it?

Reply:
when dispatching mpp tasks, in function untwistPlanAndRemoveUnionAll, all the operators will be cloned.